From 7c5c1f1b37029ada06100743f13f5acb5ad91bb9 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 11 Mar 2014 16:09:43 +0100
Subject: [PATCH 1/2] Determine devicePixelRatio from environment variable if
 set

Otherwise it would be the job of the platform plugin to set the right value. Having this
at a central place as option when the platform plugin doesn't override it is what we need.

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/gui/kernel/qplatformwindow.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gui/kernel/qplatformwindow.cpp b/src/gui/kernel/qplatformwindow.cpp
index bd5e21c..bb12ba9 100644
--- a/src/gui/kernel/qplatformwindow.cpp
+++ b/src/gui/kernel/qplatformwindow.cpp
@@ -363,6 +363,9 @@ void QPlatformWindow::handleContentOrientationChange(Qt::ScreenOrientation orien
 */
 qreal QPlatformWindow::devicePixelRatio() const
 {
+    QByteArray dpr = qgetenv("QT_PLATFORM_DPR");
+    if (dpr.length() > 0)
+        return static_cast<qreal>(dpr.toDouble());
     return 1.0;
 }
 
-- 
2.1.1

